src/components/show-edit-pod/show-edit-pod.js

```diff
@@ -64,8 +59,8 @@ class ShowEditPod extends React.Component {
       this.props.onCancel(ev);
     }
 
-    if (this.stateControlled) {
-      this.setState({ editing: false }); // eslint-disable-line react/no-unused-state
+    if (!this.isControlled) {
```

Comment:
this if statement repeats 3 times, taking for granted one is nested inside if(valid), but we could potentially reduce it to at least 2?

Reply:
Good point, I have added a method to do that.